x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Block.java

```diff
@@ -208,6 +209,24 @@ interface Builder extends BlockLoader.Builder, Releasable {
          * Builds the block. This method can be called multiple times.
          */
         Block build();
+
+        /**
+         * Build many {@link Block}s at once, releasing any partially built blocks
+         * if any fail.
+         */
+        static Block[] buildAll(Block.Builder... builders) {
```

Comment:
Nice!

Reply:
++ this will be incredibly useful, and avoid future bugs!

Reply:
Should we close the builders too, on failure?

Reply:
I was hoping we can expect the builders to be in `try` blocks.